src/CoreWCF.Http/src/CoreWCF/Channels/HttpRequestContext.cs

```diff
@@ -348,6 +348,31 @@ public AspNetCoreHttpContext(IHttpTransportFactorySettings settings, HttpContext
 
             public override string HttpMethod => _aspNetContext.Request.Method;
 
+            public override bool KeepAliveEnabled
+            {
+                get
+                {
+                    string connectionHeader = HttpRequestHeader.Connection.ToString();
+
+                    if (!_aspNetContext.Request.Headers.ContainsKey(connectionHeader))
+                    {
+                        return base.KeepAliveEnabled;
+                    }
+                    else if (StringComparer.OrdinalIgnoreCase.Equals("keep-alive", _aspNetContext.Request.Headers[connectionHeader]))
+                    {
```

Comment:
Is the binding has been configured to not use keep-alive, we should always return false otherwise the client can control resource usage on the server by creating a large number of persistent connections. I think starting with
```c#
if (!base.KeepAliveEnabled) return false;
```
That way if the binding has it disabled, we always return "close".

Reply:
I think this whole method can be simplified.
```c#
if (!base.KeepAliveEnabled) return false;
// We know base.KeepAliveEnabled is true
if (_aspNetContext.Request.Headers.ContainsKey(connectionHeader))
{
    return !StringComparer.OrdinalIgnoreCase.Equals("close", _aspNetContext.Request.Headers[connectionHeader])
}

return true;
```
I think that's sufficient. If base.KeepAliveEnabled is false, we always return false. Otherwise it's enabled so if the client provided a value, use that. If the client didn't supply a value, return true (as we know it's enabled).